File: two-pointers/is-subsequence/is-subsequence.go
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	if len(s) > len(t) {
		return false;
	}

	i, j := 0, 0
	for (i <= len(t) - 1) {
		if s[j] == t[i] {
			j++;
			if j == len(s) {
				return true
			}
		}

		i++
	}

	return false
}

File: two-pointers/is-subsequence/is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    if (s.length == 0) return true;
    if (s.length > t.length) return false;

    for (let i = 0, j = 0; i <= t.length - 1; i++)
    {
        if (s[j] == t[i])
        {
            j++;
            if (j == s.length)
            {
                return true;
            }
        }
    }

    return false;
};